pkg/controller/common/crdsRemover.go

```diff
@@ -0,0 +1,120 @@
+package common
+
+import (
+	"fmt"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	objectreferencesv1 "github.com/openshift/custom-resource-status/objectreferences/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"sync"
+)
+
+// This function creates a CRD name with the form <plural>.<group>,
+// where <plural> is the lowercase plural form of the Kind.
+// Note that this doesn't use any generic pluralization, so may not
+// work for CRDs other than the old SSP CRDs we're trying to handle here.
+func groupKindToCRDName(gk schema.GroupKind) string {
+	plural := strings.ToLower(gk.Kind) + "s"
+	return fmt.Sprintf("%s.%s", plural, gk.Group)
+}
+
+// returns true if not found or if deletion succeeded, and false otherwise.
+func removeCRD(clt client.Client, req *HcoRequest, crd schema.GroupKind) bool {
+	crdName := groupKindToCRDName(crd)
+	found := &apiextensionsv1.CustomResourceDefinition{}
+	key := client.ObjectKey{Namespace: hcoutil.UndefinedNamespace, Name: crdName}
+	err := clt.Get(req.Ctx, key, found)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			req.Logger.Error(err, fmt.Sprintf("failed to read the %s CRD; %s", crdName, err.Error()))
+			return false
+		}
+	} else {
+		err = clt.Delete(req.Ctx, found)
+		if err != nil {
+			req.Logger.Error(err, fmt.Sprintf("failed to remove the %s CRD; %s", crdName, err.Error()))
+			return false
+		}
+
+		req.Logger.Info("successfully removed CRD", "CRD Name", crdName)
+	}
+
+	return true
+}
+
+// returns a slice of CRDs that were successfully removed or don't exist, and a slice of CRDs that failed to remove.
+func RemoveCRDs(clt client.Client, req *HcoRequest, crds []schema.GroupKind) ([]schema.GroupKind, []schema.GroupKind) {
+	removed := make([]schema.GroupKind, 0, len(crds))
+	unremoved := make([]schema.GroupKind, 0, len(crds))
+
+	// The deletion is performed concurrently for all CRDs.
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+	wg.Add(len(crds))
+
+	for _, crd := range crds {
+		go func(crd schema.GroupKind) {
```

Comment:
I think it will be better to rename the function parameter **or** the loop variable, to prevent confusion.

Reply:
done